File: DestinationManager.cpp
#include "DestinationManager.h"

// Initialisierung des statischen Members
std::pair<float, float> DestinationManager::destination_ = {0.0f, 0.0f};

void DestinationManager::set_destination(const std::pair<float, float>& destination) {
    destination_ = destination;
}

std::pair<float, float> DestinationManager::get_destination() {
    return destination_;
}

File: src/DestinationManager.py
# destination_manager.py
"""
Verwaltet global gespeicherte Zielkoordinaten (Latitude, Longitude).
"""

from typing import Tuple


class DestinationManager:
    _destination: Tuple[float, float] = (0.0, 0.0)

    @classmethod
    def set_destination(cls, destination: Tuple[float, float]) -> None:
        """
        Setzt das Ziel (Breitengrad, Längengrad).

        :param destination: Tuple[float, float] mit (latitude, longitude)
        """
        cls._destination = destination

    @classmethod
    def get_destination(cls) -> Tuple[float, float]:
        """
        Gibt das aktuell gesetzte Ziel zurück.

        :return: Tuple[float, float] mit (latitude, longitude)
        """
        return cls._destination
